### ISSUE TITLE
vSphere input plugin sends the same metrics several times

### ISSUE THREAD

**Issue by VorobevPavel-dev** [influxdata/telegraf#9088]
### Description
In our monitoring system there is a need for monitor vSphere using ```[[inputs.vsphere]]```. We use Telegraf v 1.11.5 (we cannot upgrade due to changes in ```[[outputs.kafka]]``` in next versions). ```usage_latest``` and ```capacity_latest``` metrics are gathered once in 300 seconds. There are no drops or losses seen in log file or in data from ```[[inputs.internal]]```. However, in Grafana dashboard there are missing values.    

### Relevant configs
/etc/telegraf/telegraf.d/inputs_vsphere.conf
``` toml

[[inputs.vsphere]]
        name_override = "service.vcenter"
        vcenters = [ "some-server/sdk" ]
        username = "username"
        password = "password"
        host_metric_exclude = ["*"]
        vm_metric_exclude = ["*"]
        datastore_metric_include = [
                                    "disk.used.latest",
                                    "disk.capacity.latest",
                                   ]
        cluster_metric_exclude = ["*"]
        datacenter_metric_exclude = ["*"]
        max_query_objects = 256
        max_query_metrics = 256
        insecure_skip_verify = true
        interval = "300s"
        collect_concurrency = 1
        discover_concurrency = 1
[inputs.vsphere.tags]
        interval = "300s"
```

/etc/telegraf/telegraf.d/outputs_kafka.confa
``` toml

[[outputs.kafka]]
  brokers = ["kafka-1:9092","kafka-2:9092","kafka-3:9092"]
  topic = "telegraf-tagged"
  version = "2.5.0"
  compression_codec = 3
  required_acks = 1
  max_message_bytes = 10000000
  data_format = "graphite"
  graphite_tag_support = true
```

### System info
- Telegraf v1.11.5
- Kafka v2.5.0

### Expected behavior
No missing points on grafana dashboard

### Actual behavior
Metrics gathered once in 10-30 minutes, not in 5 min. It causes empty spaces in Grafana. 
![image_2021-03-31_10-34-41](https://user-images.githubusercontent.com/66829073/113538908-18bcae00-95e5-11eb-8093-371599bc20dd.png)

Futhermore not all points were sent to Kafka but there are same points (got with kafka-console-consumer + grep. New same metric appears every 3-5 minutes):
``` toml
service.vcenter.capacity_latest;<tags> 21474574336 1617266100
service.vcenter.capacity_latest;<tags same as above> 21474574336 1617266100
service.vcenter.capacity_latest;<tags same as above> 21474574336 1617266100
service.vcenter.capacity_latest;<tags same as above> 21474574336 1617266100
```
Moreover, buffer is never empty and stays at 200k metrics, does not overflow.

#### Log
```2021-04-05T04:40:07Z D! [inputs.vsphere] Raw interval 5m5.014564365s, padded: 7m35.014564365s, estimated: 5m0s
2021-04-05T04:40:07Z D! [inputs.vsphere] Interval estimated to 5m0s
2021-04-05T04:40:07Z D! [inputs.vsphere]: Latest: 2021-04-05 04:30:00 +0000 UTC, elapsed: 612.586834, resource: datastore
2021-04-05T04:40:07Z D! [inputs.vsphere]: Collecting metrics for 64 objects of type datastore for host-1
2021-04-05T04:40:07Z D! [inputs.vsphere]: Queuing query: 64 objects, 128 metrics (0 remaining) of type datastore for host-1. Total objects 64 (final chunk)
2021-04-05T04:40:07Z D! [inputs.vsphere] Query for datastore has 64 QuerySpecs
2021-04-05T04:40:07Z D! [inputs.vsphere] Query for datastore returned metrics for 64 objects
2021-04-05T04:40:07Z D! [inputs.vsphere] CollectChunk for datastore returned 136 metrics
2021-04-05T04:40:07Z D! [inputs.vsphere] Latest sample for datastore set to 2021-04-05 04:35:00 +0000 UTC
2021-04-05T04:40:07Z D! [inputs.vsphere] Purged timestamp cache. 16 deleted with 48 remaining
2021-04-05T04:40:07Z D! [inputs.vsphere] Query for vm returned metrics for 256 objects
2021-04-05T04:40:07Z D! [inputs.vsphere] Raw interval 22.357014689s, padded: 32.357014689s, estimated: 20s
2021-04-05T04:40:07Z D! [inputs.vsphere] Interval estimated to 20s
2021-04-05T04:40:07Z D! [inputs.vsphere]: Latest: 2021-04-05 04:39:40 +0000 UTC, elapsed: 32.932816, resource: host
2021-04-05T04:40:07Z D! [inputs.vsphere]: Collecting metrics for 177 objects of type host for host-1
2021-04-05T04:40:07Z D! [inputs.vsphere]: Queuing query: 177 objects, 4425 metrics (0 remaining) of type host for host-1. Total objects 177 (final chunk)
2021-04-05T04:40:07Z D! [inputs.vsphere] Query for host has 177 QuerySpecs
2021-04-05T04:40:07Z D! [inputs.vsphere] Query for vm returned metrics for 145 objects
2021-04-05T04:40:07Z D! [inputs.vsphere] Query for vm returned metrics for 257 objects
2021-04-05T04:40:08Z D! [inputs.vsphere] CollectChunk for vm returned 13040 metrics
2021-04-05T04:40:08Z D! [inputs.vsphere] Query for vm has 256 QuerySpecs
2021-04-05T04:40:08Z D! [inputs.vsphere] CollectChunk for vm returned 24506 metrics
2021-04-05T04:40:08Z D! [inputs.vsphere] Query for vm has 256 QuerySpecs
2021-04-05T04:40:08Z D! [inputs.vsphere] CollectChunk for vm returned 25034 metrics
2021-04-05T04:40:08Z D! [inputs.vsphere] Query for vm has 256 QuerySpecs
2021-04-05T04:40:08Z D! [outputs.kafka] wrote batch of 12000 metrics in 2.240476437s
2021-04-05T04:40:08Z D! [inputs.vsphere] Query for host returned metrics for 177 objects
2021-04-05T04:40:08Z D! [inputs.vsphere] Query for vm returned metrics for 256 objects
2021-04-05T04:40:08Z D! [inputs.vsphere] Query for vm returned metrics for 256 objects
2021-04-05T04:40:08Z D! [inputs.vsphere] Query for vm returned metrics for 256 objects
2021-04-05T04:40:08Z D! [inputs.vsphere] CollectChunk for host returned 26409 metrics
2021-04-05T04:40:08Z D! [inputs.vsphere] Latest sample for host set to 2021-04-05 04:40:00 +0000 UTC
2021-04-05T04:40:08Z D! [inputs.vsphere] Purged timestamp cache. 0 deleted with 177 remaining
2021-04-05T04:40:08Z D! [inputs.vsphere] CollectChunk for vm returned 24898 metrics
2021-04-05T04:40:08Z D! [inputs.vsphere] Query for vm has 256 QuerySpecs
2021-04-05T04:40:08Z D! [inputs.vsphere] CollectChunk for vm returned 23044 metrics
2021-04-05T04:40:08Z D! [inputs.vsphere] Query for vm has 256 QuerySpecs
2021-04-05T04:40:09Z D! [inputs.vsphere] CollectChunk for vm returned 24884 metrics
2021-04-05T04:40:09Z D! [inputs.vsphere] Query for vm has 256 QuerySpecs
```

**Comment by VorobevPavel-dev** [comment]
Please take a look, @prydin, @danielnelson

**Comment by prydin** [comment]
Disk capacity is only updated once ever 30 minutes in vCenter, so this is the expected behavior.

**Comment by reimda** [comment]
1.11.5 was released in August 2019 and not supported any more.  If we found a problem in 1.11.5 vsphere and fixed it, the fix would be released in a future release (1.18.2 or later.)  You need to test on the latest release before you can expect a fix.

If you have a problem with kafka that prevents upgrade, maybe you should test vsphere on a current release with a different output (like output.file) just to see if the vsphere problem is in the current release.

Have you filed an issue for the kafka issue that's preventing upgrade?  If so, could you link it here?

**Comment by VorobevPavel-dev** [comment]
I think that option with ouput not in kafka is not really possible in our environment. However we will try to gather metrics once in half an hour. If it fix our problem i'll close an issue. Thank you for your quick responce
